frontend/src/DialectReportView.tsx

```diff
@@ -16,66 +16,71 @@ export const DialectReportView = ({
   allImplementationsData,
 }: {
   reportData: ReportData;
-  allImplementationsData?: Record<string, Implementation>;
+  allImplementationsData: Record<string, Implementation>;
 }) => {
   const params = useSearchParams();
 
   const languages = useMemo(() => {
     const langs = Array.from(reportData.implementations.values()).map(
-      (impl) => impl.metadata.language,
+      (impl) => impl.metadata.language
     );
     return Array.from(new Set(langs).values());
   }, [reportData]);
 
-  const filteredData = useMemo(() => {
-    const filteredReportData = { ...reportData };
-    let filteredOtherImplementationsData: Record<string, Implementation> = {};
+  const filteredReportData = useMemo(() => {
+    const filteredData = { ...reportData };
     const selectedLanguages = params.getAll("language");
 
     if (selectedLanguages.length > 0) {
       const filteredReportArray = Array.from(
-        reportData.implementations.entries(),
+        reportData.implementations.entries()
       ).filter(([, data]) =>
-        selectedLanguages.includes(data.metadata.language),
+        selectedLanguages.includes(data.metadata.language)
       );
       const filteredReportImplementationsMap = new Map(filteredReportArray);
-      filteredReportData.implementations = filteredReportImplementationsMap;
-      if (allImplementationsData) {
-        filteredOtherImplementationsData = filterOtherImplementations(
-          allImplementationsData,
-          selectedLanguages,
-          filteredReportImplementationsMap,
-        );
-      }
-    } else {
+      filteredData.implementations = filteredReportImplementationsMap;
+    }
+
+    return filteredData;
+  }, [reportData, params]);
+
+  const filteredOtherImplementationsData = useMemo(() => {
+    let filteredData: Record<string, Implementation> = {};
+    const selectedLanguages = params.getAll("language");
+
+    if (selectedLanguages.length > 0) {
```

Comment:
That just does the same work twice. You did the work while calculating `filteredReportData` and you can reuse it here. It simplifies thing greatly. I believe the whole useMemo logic can just become:
```ts
    const availableLanguages = selectedLanguages.length ? selectedLanguages : languages
    return filterOtherImplementations(
      allImplementationsData,
      availableLanguages,
      filteredReportData.implementations,
    );
```
But again, please check if I didn't miss anything and if it will work correctly :)

Reply:
Yes! I should've catched this! Sorry!